midterm/src/Midterm/Restaurants.java

```diff
@@ -0,0 +1,110 @@
+package Midterm;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+
+/**
+ * Servlet implementation class Restaurants
+ */
+@WebServlet("/suggest/restaurants/")
+public class Restaurants extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+	String[]url ;
+	int id[] ;
+	String name[];
+	int [] designRatings ;
+	int [] tasteRatings ;
+	RandomRestaurant tempid;
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+	public void init()
+    {
+    	String[] url=new String[22];
+    	int[] id=new int[22];
+    	String[] name=new String[22];
+    	int[] designRatings=new int[22];
+    	int[] tasteRatings=new int[22];
```

Comment:
This assumption of array of 22 items is wrong. Ratings can be submitted by anyone and therefore it is wrong.